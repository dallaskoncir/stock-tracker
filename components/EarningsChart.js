import PropTypes from 'prop-types';
import Highcharts from 'highcharts';
import HighchartsReact from 'highcharts-react-official';

export default function EarningsChart({ name, annualEarnings }) {
    const mappedEarningsData = annualEarnings.map((data) => {
        const { period, v } = data;
        const formattedDate = Date.parse(period);

        return [formattedDate, v];
    });

    const options = {
        constructorType: 'stockChart',
        series: [
            {
                name,
                data: mappedEarningsData.reverse()
            }
        ],
        title: {
            text: 'Annual Earnings Per Share'
        },
        xAxis: {
            labels: {
                format: '{value:%Y-%m-%d}',
                rotation: 45,
                align: 'left'
            },
            tickPixelInterval: 50,
            type: 'datetime'
        }
    };

    return <HighchartsReact highcharts={Highcharts} options={options} />;
}

EarningsChart.propTypes = {
    name: PropTypes.string.isRequired,
    annualEarnings: PropTypes.array.isRequired
};
