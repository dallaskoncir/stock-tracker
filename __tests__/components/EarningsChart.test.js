import { render } from '@testing-library/react';

import EarningsChart from '../../components/EarningsChart';

describe('<EarningsChart />', () => {
    test('Renders correctly with annualEarnings data supplied', () => {
        const annualEarnings = [
            { period: '2020-09-26', v: 3.2753 },
            { period: '2019-09-28', v: 2.9714 }
        ];
        const { container } = render(
            <EarningsChart name="Apple Inc." annualEarnings={annualEarnings} />
        );

        expect(container).toMatchSnapshot();
    });
});
