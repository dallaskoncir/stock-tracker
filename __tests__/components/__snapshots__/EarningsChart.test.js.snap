// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EarningsChart /> Renders correctly with annualEarnings data supplied 1`] = `
<div>
  <div
    data-highcharts-chart="0"
    style="overflow: hidden;"
  >
    <div
      class="highcharts-container "
      dir="ltr"
      id="highcharts-cp9jvk4-0"
      style="position: relative; overflow: hidden; width: 600px; height: 400px; text-align: left; line-height: normal; z-index: 0; user-select: none;"
    >
      <svg
        class="highcharts-root"
        height="400"
        style="font-family:\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Arial, Helvetica, sans-serif;font-size:12px;"
        version="1.1"
        viewBox="0 0 600 400"
        width="600"
        xmlns="http://www.w3.org/2000/svg"
      >
        <desc>
          Created with Highcharts 9.0.0
        </desc>
        <defs>
          <clippath
            id="highcharts-cp9jvk4-1-"
          >
            <rect
              fill="none"
              height="NaN"
              width="NaN"
              x="0"
              y="0"
            />
          </clippath>
          <clippath
            id="highcharts-cp9jvk4-4-"
          >
            <rect
              fill="none"
              height="400"
              width="NaN"
              x="-99"
              y="-10"
            />
          </clippath>
          <clippath
            id="highcharts-cp9jvk4-5-"
          >
            <rect
              fill="none"
              height="NaN"
              width="NaN"
              x="0"
              y="0"
            />
          </clippath>
        </defs>
        <rect
          class="highcharts-background"
          fill="#ffffff"
          height="400"
          rx="0"
          ry="0"
          width="600"
          x="0"
          y="0"
        />
        <rect
          class="highcharts-plot-background"
          fill="none"
          height="NaN"
          width="NaN"
          x="NaN"
          y="10"
        />
        <g
          class="highcharts-grid highcharts-xaxis-grid"
          data-z-index="1"
        >
          <path
            class="highcharts-grid-line"
            d="M 0 0"
            data-z-index="1"
            fill="none"
            opacity="1"
          />
        </g>
        <g
          class="highcharts-grid highcharts-yaxis-grid"
          data-z-index="1"
        >
          <path
            class="highcharts-grid-line"
            d="M 0 0"
            data-z-index="1"
            fill="none"
            opacity="1"
            stroke="#e6e6e6"
            stroke-width="1"
          />
          <path
            class="highcharts-grid-line"
            d="M 0 0"
            data-z-index="1"
            fill="none"
            opacity="1"
            stroke="#e6e6e6"
            stroke-width="1"
          />
          <path
            class="highcharts-grid-line"
            d="M 0 0"
            data-z-index="1"
            fill="none"
            opacity="1"
            stroke="#e6e6e6"
            stroke-width="1"
          />
          <path
            class="highcharts-grid-line"
            d="M 0 0"
            data-z-index="1"
            fill="none"
            opacity="1"
            stroke="#e6e6e6"
            stroke-width="1"
          />
          <path
            class="highcharts-grid-line"
            d="M 0 0"
            data-z-index="1"
            fill="none"
            opacity="1"
            stroke="#e6e6e6"
            stroke-width="1"
          />
        </g>
        <rect
          class="highcharts-plot-border"
          data-z-index="1"
          fill="none"
          height="0"
          width="0"
          x="0"
          y="10"
        />
        <g
          class="highcharts-axis highcharts-xaxis"
          data-z-index="2"
        >
          <path
            class="highcharts-tick"
            d="M 0 0"
            fill="none"
            opacity="1"
            stroke="#ccd6eb"
            stroke-width="1"
          />
          <path
            class="highcharts-axis-line"
            d="M 0 0"
            data-z-index="7"
            fill="none"
            stroke="#ccd6eb"
            stroke-width="1"
          />
        </g>
        <g
          class="highcharts-axis highcharts-yaxis"
          data-z-index="2"
        >
          <text
            class="highcharts-axis-title"
            data-z-index="7"
            style="color:#666666;fill:#666666;"
            text-anchor="middle"
            transform="translate(0,0) rotate(270 0 -9999)"
            x="0"
            y="-9999"
          >
            Values
          </text>
          <path
            class="highcharts-axis-line"
            d="M 0 0"
            data-z-index="7"
            fill="none"
          />
        </g>
        <g
          class="highcharts-series-group"
          data-z-index="3"
        >
          <g
            class="highcharts-series highcharts-series-0 highcharts-line-series highcharts-color-0"
            clip-path="url(#highcharts-cp9jvk4-5-)"
            data-z-index="0.1"
            opacity="1"
            transform="translate(0,10) scale(1 1)"
          >
            <path
              class="highcharts-graph"
              d="M -100000 -100000 L -100000 -100000"
              data-z-index="1"
              fill="none"
              stroke="#7cb5ec"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              class="highcharts-tracker-line"
              d="M -100000 -100000 L -100000 -100000"
              data-z-index="2"
              fill="none"
              stroke="rgba(192,192,192,0.002)"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="22"
              visibility="visible"
            />
          </g>
          <g
            class="highcharts-markers highcharts-series-0 highcharts-line-series highcharts-color-0 highcharts-tracker"
            clip-path="url(#highcharts-cp9jvk4-4-)"
            data-z-index="0.1"
            opacity="1"
            transform="translate(0,10) scale(1 1)"
          />
        </g>
        <text
          class="highcharts-title"
          data-z-index="4"
          style="color:#333333;font-size:18px;fill:#333333;"
          text-anchor="middle"
          x="300"
          y="24"
        >
          Annual Earnings Per Share
        </text>
        <text
          class="highcharts-subtitle"
          data-z-index="4"
          style="color:#666666;fill:#666666;"
          text-anchor="middle"
          x="300"
          y="10"
        />
        <text
          class="highcharts-caption"
          data-z-index="4"
          style="color:#666666;fill:#666666;"
          text-anchor="start"
          x="10"
          y="397"
        />
        <g
          class="highcharts-legend"
          data-z-index="7"
          transform="translate(300,362)"
        >
          <rect
            class="highcharts-legend-box"
            fill="none"
            rx="0"
            ry="0"
            visibility="visible"
          />
          <g
            data-z-index="1"
          >
            <g>
              <g
                class="highcharts-legend-item highcharts-line-series highcharts-color-0 highcharts-series-0"
                data-z-index="1"
                transform="translate(8,3)"
              >
                <path
                  class="highcharts-graph"
                  d="M 0 11 L 16 11"
                  fill="none"
                  stroke="#7cb5ec"
                  stroke-width="2"
                />
                <path
                  class="highcharts-point"
                  d="M 8 15 A 4 4 0 1 1 8.003999999333336 14.999998000000167 Z"
                  fill="#7cb5ec"
                  opacity="1"
                />
                <text
                  data-z-index="2"
                  style="color:#333333;cursor:pointer;font-size:12px;font-weight:bold;fill:#333333;"
                  text-anchor="start"
                  x="21"
                  y="15"
                >
                  Apple Inc.
                </text>
              </g>
            </g>
          </g>
        </g>
        <g
          class="highcharts-axis-labels highcharts-xaxis-labels"
          data-z-index="7"
        >
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="start"
            transform="translate(0,0) rotate(45 0 0)"
            x="0"
          >
            2020-01-01
          </text>
        </g>
        <g
          class="highcharts-axis-labels highcharts-yaxis-labels"
          data-z-index="7"
        >
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="end"
            transform="translate(0,0)"
            x="0"
          >
            2.9
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="end"
            transform="translate(0,0)"
            x="0"
          >
            3
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="end"
            transform="translate(0,0)"
            x="0"
          >
            3.1
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="end"
            transform="translate(0,0)"
            x="0"
          >
            3.2
          </text>
          <text
            style="color:#666666;cursor:default;font-size:11px;fill:#666666;"
            text-anchor="end"
            transform="translate(0,0)"
            x="0"
          >
            3.3
          </text>
        </g>
        <text
          class="highcharts-credits"
          data-z-index="8"
          style="cursor:pointer;color:#999999;font-size:9px;fill:#999999;"
          text-anchor="end"
          x="590"
          y="395"
        >
          Highcharts.com
        </text>
      </svg>
    </div>
  </div>
</div>
`;
